docs/source/index.rst

```diff
@@ -1,20 +1,204 @@
+from an OrientDB database, it is meant as a generic schema info generation example.
+See the target database homepage for schema generation instructions.
+
+.. code:: python
+
+    from graphql_compiler import (
+        get_graphql_schema_from_orientdb_schema_data
+    )
+    from graphql_compiler.schema_generation.orientdb.utils import ORIENTDB_SCHEMA_RECORDS_QUERY
+
+    client = your_function_that_returns_an_orientdb_client()
+    schema_records = client.command(ORIENTDB_SCHEMA_RECORDS_QUERY)
+    schema_data = [record.oRecordData for record in schema_records]
+    schema, type_equivalence_hints = get_graphql_schema_from_orientdb_schema_data(schema_data)
+
+At the core of generated schema info is the GraphQL :code:`schema`. The database might be
+reflected in the :code:`schema` as follows:
+
+.. code::
+
+    type Animal {
+        name: String
+        out_Animal_LivesIn: [Continent]
+    }
+
+    type Continent {
+        name: String
+        in_AnimalLivesIn: [Animal]
+    }
+
+In the :code:`schema` above:
+
+-   :code:`Animal` represents a non-abstract vertex. For relational databases, we think of
+    tables as the non-abstract vertices.
+-   :code:`name` is a **property field** and :code:`out_Animal_LivesIn` is a **vertex field**,
+    which represent an outbound edge to a vertex in the graph. For graph databases,
+    edges can be reflected from the database schema. However, for relational databases edges have
+    to be manually specified. See :doc:`SQL <databases/sql>` for more information.
+
+Query Compilation and Execution
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once we have the schema info we can write the following query to get the names of all the animals
+that live in Africa:
+
+.. code:: python
+
+    graphql_query = """
+    {
+        Animal {
+            name @output(out_name: "animal_name")
+            out_Animal_LivesIn {
+                name @filter(op_name: "=", value: ["$continent"])
+            }
+        }
+    }
+    """
+    parameters = {'continent': 'Africa'}
+
+There are a couple of things to notice about queries:
 
+- All queries start with a vertex and branch to other vertices using **vertex fields**.
+- **Directives**, such as :code:`@output` and :code:`@filter`, specify the semantics of a query.
+  :code:`@output` indicates which properties to emit and :code:`@filter` specifies a filter
+  operation.
 
+With the query at hand, we can compile it and get the corresponding results from OrientDB.
 
-Indices and tables
-==================
+.. code:: python
+
+    from graphql_compiler import graphql_to_match
+
+    compilation_result = graphql_to_match(
+        schema, graphql_query, parameters, type_equivalence_hints)
+
+    # Execute query assuming a pyorient client. Other clients may have a different interface.
+    print([result.oRecordData for result in client.query(query)])
+    # [{'animal_name': 'Elephant'}, {'animal_name': 'Lion'}, ...]
+
+What's Next?
```

Comment:
I'm not sure if this is the best title for this section, since it doesn't exist any specific follow up actions by the user nor us. We could probably get rid of it or replace it with something more accurate.

Reply:
In restructuredtext you cannot skip section headers.  
https://stackoverflow.com/questions/33149169/skipping-heading-levels-in-restructuredtext
So even if this header doesn't say much it helps as an intermediate header.

Reply:
I think you are confusing content problems and technical limitations. If the title were more appropriate, the fact that you cannot skip heading levels would be irrelevant – because the heading would be actually useful.

Reply:
Any recommendations for a better title? I can't think of a substantially better one.

Reply:
We just synced about this. A few things:
- For now it's fine, but we want to make sure the Home Page doesn't grow out of scope / get too lengthy. It would be nice to have to have a nav for Home either way like here: https://mkdocs.readthedocs.io/en/stable/ From our chat it sounded like there might be some bugs, so let's leave some comments about next steps.
- As for a better title, it would be something along the lines of `Features`.

Reply:
I added a TODO above to the restructuring and the nav bar at the top of the page. 

I renamed the section to `Features`. 

I left `Advanced Features` as `Advanced Features` instead of `Advanced` since  `Advanced Features` also appears in the nav bar where the word `Features` is not mentioned.